src/ember-app.js

```diff
@@ -310,16 +289,7 @@ class EmberApp {
       })
       .catch(error => result.error = error)
       .then(() => result._finalize())
-      .finally(() => {
-        if (instance && !result.instanceDestroyed) {
-          result.instanceDestroyed = true;
-          instance.destroy();
-
-          if (destroyAppInstanceTimer) {
-            clearTimeout(destroyAppInstanceTimer);
-          }
-        }
-      });
+      .finally(() => result._destroyAppInstance(destroyAppInstanceInMs));
```

Comment:
This is not the desired behavior that we want. We want to start a timer before the `visit` request starts in order to make sure we destroy the instance forcefully if the request never completes.

With your refactor, we will start the timer after the `visit` request is completed which might not be of any use.